StitchPanorama.py

```diff
@@ -1,77 +1,118 @@
 import cv2
 import numpy as np
+import sys
+from matcher import matcher
 #https://python.plainenglish.io/opencv-image-stitching-second-part-388784ccd1a
 #https://towardsdatascience.com/image-panorama-stitching-with-opencv-2402bde6b46c
 class StitchPanorama:
-    #def __init__(self):
-    # img_ = previous background image
-    # img = current one
-    def stitch(self, img_, img):
-        img1 = cv2.cvtColor(img_,cv2.COLOR_BGR2GRAY)
-        img2 = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
 
-        sift = cv2.SIFT_create()
-        #sift = cv2.createStitcher() if imutils.is_cv3() else cv2.Stitcher_create()
-        # find key points
-        kp1, des1 = sift.detectAndCompute(img1,None)
-        kp2, des2 = sift.detectAndCompute(img2,None)
+    def __init__(self, frames):
+        self.images = frames
+        self.count = len(self.images)
+        self.left_list, self.right_list, self.center_im = [], [],None
+        self.matcher_obj = matcher()
+        self.prepare_lists()
 
-        #cv2.imshow('original_image_left_keypoints',cv2.drawKeypoints(img_,kp1,None))
 
-        #FLANN_INDEX_KDTREE = 0
-        #index_params = dict(algorithm = FLANN_INDEX_KDTREE, trees = 5)
-        #search_params = dict(checks = 50)
-        #match = cv2.FlannBasedMatcher(index_params, search_params)
-        match = cv2.BFMatcher()
-        matches = match.knnMatch(des1,des2,k=2)
+    def simpleStitch(self):
+        stitchy=cv2.Stitcher.create()
+        ret, panorama = stitchy.stitch(self.images)
+        print(cv2.STITCHER_OK, ret)
+        if ret != cv2.STITCHER_OK:
+            print('stitch Failed! error: ', ret)
+        else:
+            return panorama
 
-        good = []
-        for m,n in matches:
-            if m.distance < 0.03*n.distance:
-                good.append(m)
+    def prepare_lists(self):
+        self.centerIdx = self.count/2
+        self.center_im = self.images[int(self.centerIdx)]
+        for i in range(self.count):
```

Comment:
I think there should be a way to avoid using for loop, like self.images[: self.centerIndex]

Reply:
but every frame is uniq one so you still need a for loop within it like self.images[frames[i], for i in range(self.centerIndex)]

Reply:
self.left_list = self.images[:self.centerIndex + 1]